File: server/Server/Models/Domain/DataSource.cs
using server.Models.Domain.Enums;

namespace server.Models.Domain;

public class DataSource
{
    public int DataSourceId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DataSourceType SourceType { get; set; } = DataSourceType.OTHER;
    public string? ConnectionDetails { get; set; } // TODO: Store as JSON string; map to complex type later if needed
    public DataSourceStatus Status { get; set; }
    public bool IsDefault { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Foreign Key to Project
    public int ProjectId { get; set; }

    // Navigation Properties
    public virtual Project Project { get; set; } = null!;
    public virtual ICollection<Asset> Assets { get; set; } = [];
}

File: frontend/src/types/dataSource/dataSource.ts
export enum DataSourceType {
    MINIO_BUCKET = 'MINIO_BUCKET',
    S3_BUCKET = 'S3_BUCKET',
    GSC_BUCKET = 'GSC_BUCKET',
    AZURE_BLOB_STORAGE = 'AZURE_BLOB_STORAGE',
    LOCAL_DIRECTORY = 'LOCAL_DIRECTORY',
    DATABASE = 'DATABASE',
    API = 'API',
    OTHER = 'OTHER',
}

export enum DataSourceStatus {
    ACTIVE = 'ACTIVE',
    INACTIVE = 'INACTIVE',
    SYNCING = 'SYNCING',
    ERROR = 'ERROR',
    ARCHIVED = 'ARCHIVED',
}

export interface DataSource {
    id: number;
    name: string;
    description?: string;
    sourceType: DataSourceType;
    status: DataSourceStatus;
    isDefault: boolean;
    createdAt: string;
    projectId: number;
    assetCount?: number;
}